src/NServiceBus.Msmq.AcceptanceTests/Distributor/When_replying_to_message_sent_from_worker.cs

```diff
@@ -0,0 +1,134 @@
+                .Done(c => c.WorkerReceivedReply)
+                .Run();
+
+            Assert.IsFalse(context.DistributorReceivedReply);
+            Assert.IsTrue(context.WorkerReceivedReply);
+        }
+
+        class Context : DistributorEndpointTemplate.DistributorContext
+        {
+            public bool DistributorReceivedReply { get; set; }
+            public bool WorkerReceivedReply { get; set; }
+        }
+
+        class Worker : EndpointConfigurationBuilder
+        {
+            public Worker()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.EnlistWithDistributor(typeof(Distributor));
+                })
+                .AddMapping<WorkerMessage>(typeof(ReplyingEndpoint));
+            }
+
+            class DispatchWorkerMessageHandler : IHandleMessages<DispatchWorkerMessage>
+            {
+                public Task Handle(DispatchWorkerMessage message, IMessageHandlerContext context)
+                {
+                    return context.Send(new WorkerMessage());
+                }
+            }
+
+            class ReplyMessageHandler : IHandleMessages<ReplyMessage>
+            {
+                Context testContext;
+
+                public ReplyMessageHandler(Context context)
+                {
+                    testContext = context;
+                }
+
+                public Task Handle(ReplyMessage message, IMessageHandlerContext context)
+                {
+                    testContext.WorkerReceivedReply = true;
+                    return Task.CompletedTask;
+                }
+            }
+        }
+
+        class Distributor : EndpointConfigurationBuilder
+        {
+            public Distributor()
+            {
+                EndpointSetup<DistributorEndpointTemplate>()
+                    .AddMapping<DispatchWorkerMessage>(typeof(Worker));
+            }
+
+            class ReplyMessageHandler : IHandleMessages<ReplyMessage>
+            {
+                Context testContext;
+
+                public ReplyMessageHandler(Context context)
+                {
+                    testContext = context;
+                }
+
+                public Task Handle(ReplyMessage message, IMessageHandlerContext context)
+                {
+                    testContext.DistributorReceivedReply = true;
+                    return Task.CompletedTask;
+                }
+            }
+        }
+
+        class ReplyingEndpoint : EndpointConfigurationBuilder
+        {
+            public ReplyingEndpoint()
+            {
+                EndpointSetup<DefaultServer>();
+            }
+
+            class WorkerMessageHandler : IHandleMessages<WorkerMessage>
+            {
+                public Task Handle(WorkerMessage message, IMessageHandlerContext context)
+                {
+                    return context.Reply(new ReplyMessage());
+                }
+            }
+        }
+
+        class DispatchWorkerMessage : ICommand
```

Comment:
they have to be public on the shared att tests because of certain downstream serializers. This isn't really necessary here and it's also not enforced.